design/proposals/variable_provision.md

```diff
@@ -0,0 +1,295 @@
+# Solution Design - Variable Provisioning
+
+## Table of Contents
+
+- [Glossary](#glossary)
+- [Useful links](#useful-links)
+- [Background](#background)
+- [Issue description](#issue-description)
+  * ['Browsing-up` for secrets management](#-browsing-up--for-secrets-management)
+  * [Secrets Domain Knowledge Required](#secrets-domain-knowledge-required)
+  * [Human Intervention Required](#human-intervention-required)
+  * [Credential Provenance](#credential-provenance)
+- [Solution](#solution)
+  * [Design](#design)
+  * [Backwards compatibility](#backwards-compatibility)
+  * [Performance](#performance)
+  * [Affected Components](#affected-components)
+- [Security](#security)
+- [Test Plan](#test-plan)
+- [Logs](#logs)
+  * [Audit](#audit)
+- [Documentation](#documentation)
+- [Version update](#version-update)
+- [Open questions](#open-questions)
+- [Breaking](#breaking)
+
+
+## Glossary
+[//]: # (Describe terms that will be used throughout the design)
+[//]: # (You can use this tool to generate a table - https://www.tablesgenerator.com/markdown_tables#)
+
+> TBD
+
+| **Term** | **Description** |
+|----------|-----------------|
+|          |                 |
+|          |                 |
+
+## Useful links
+[//]: # (Add links that may be useful for the reader)
+
+> TBD
+
+## Background
+
+The purpose of Variables in Conjur is to put values in them for applications to
+use for service access. These values often consist of:
+
+- Passwords
+- API Keys
+- x509 Private Keys
+- x509 Certificates
+
+The typical workflow for a Variable is:
+
+- The Variable is created when it is defined in a Conjur policy document and loaded
+  into Conjur:
+
+  ```YAML
+  # my-variable.yaml
+
+  - !variable my-api-key
+  ```
+
+  ```sh-session
+  $ conjur policy load root my-variable.yaml
+  ...
+  ```
+
+- The owner of the value uses the Conjur API or CLI to store the value in the Variable:
+  
+  ```sh-session
+  $ conjur variable values add my-api-key abcdef12345
+  ...
+  ```
+
+- Once stored, access to the variable may be granted to applications, and those
+  applications can use value using the Conjur API, Summon, Secretless, etc.
+
+## Issue description
+
+There are a couple of issues this solution intends to address:
+
+### 'Browsing-up` for secrets management
+
+"Browse-up" is defined as [[ref]](https://www.ncsc.gov.uk/whitepaper/security-architecture-anti-patterns#section_3):
+
+> When administration of a system is performed from a device which is less trusted
+> than the system being administered.
+
+Unless performed only on dedicated, controlled workspaces, the `variable values add`
+step in our existing workflow necessitates that all secrets stored in Conjur (a
+"trusted system") must first travel through a user's workstation (a "less trusted system").
+
+### Secrets Domain Knowledge Required
```

Comment:
Not sure if it's part of this section, but generating certificates for the K8s Authenticator seems a lot more complex due by asking customers to generate these certificates.

Reply:
Yes, this would completely eliminate the need for certificate creation (either manually, or the rake task). The parameters for the certificates would just become a part of the template for the authenticator policy.

I actually used this from the prototype in one of the openshift operator demos to completely eliminate that step from the openshift follower installation flow.